### ISSUE TITLE
minikube has `kube-proxy` in `Blocked` state

### ISSUE THREAD

**Issue by surajssd** [kubernetes/minikube#3770]
### How to replicate the error, including the exact command-lines used.

```console
minikube start \
--kubernetes-version v1.13.3 \
--cpus 3 --memory 3000  \
--extra-config=apiserver.authorization-mode=RBAC \
--extra-config=apiserver.enable-admission-plugins=Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota \
--v 10
```

### The full output of the command that failed

See that the `kube-proxy-r28bc` and the `storage-provisioner` pods are in blocked state.

```console
$ kubectl get pods --all-namespaces
NAMESPACE     NAME                               READY   STATUS    RESTARTS   AGE
...
kube-system   kube-proxy-r28bc                   0/1     Blocked   0          10m
kube-system   kube-scheduler-minikube            1/1     Running   0          10m
kube-system   storage-provisioner                0/1     Blocked   0          10m
```

So I checked why this is blocked, seems like the host networking is blocked on this node for the kinda of host networking kube-proxy is demanding.

```yaml
$ kubectl get pods -n kube-system kube-proxy-r28bc -o yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    scheduler.alpha.kubernetes.io/critical-pod: ""
  creationTimestamp: "2019-02-28T07:52:24Z"
  generateName: kube-proxy-
  labels:
    controller-revision-hash: 84cfdd5c48
    k8s-app: kube-proxy
    pod-template-generation: "1"
  name: kube-proxy-r28bc
...
  containerStatuses:
  - image: k8s.gcr.io/kube-proxy:v1.13.2
    imageID: ""
    lastState: {}
    name: kube-proxy
    ready: false
    restartCount: 0
    state:
      waiting:
        reason: Blocked
  hostIP: 192.168.123.217
  message: pod with UID "c907ef26-3b2d-11e9-8e8f-d04e73cc09f7" specified host networking,
    but is disallowed
  phase: Pending
  podIP: 192.168.123.217
  qosClass: BestEffort
  reason: Forbidden
  startTime: "2019-02-28T07:52:24Z"
```

Checked on the minikube node:

```console
# systemctl status kubelet
● kubelet.service - kubelet: The Kubernetes Node Agent
   Loaded: loaded (/usr/lib/systemd/system/kubelet.service; enabled; vendor preset: enabled)
  Drop-In: /etc/systemd/system/kubelet.service.d
           └─10-kubeadm.conf
   Active: active (running) since Thu 2019-02-28 07:51:45 UTC; 29min ago
     Docs: http://kubernetes.io/docs/
 Main PID: 2738 (kubelet)
    Tasks: 19 (limit: 3133)
   CGroup: /system.slice/kubelet.service
           └─2738 /usr/bin/kubelet --pod-manifest-path=/etc/kubernetes/manifests --allow-privileged=true --cluster-dns=10.96.0.10 --authorization-mode=Webhook --client-ca-file=/var/lib/minikube/certs/ca.crt --cgroup-driver=cgroupfs --kubeconfig=/etc/kubernetes/kubelet.conf --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --cluster-domain=cluster.local --host-network-sources=file --container-runtime=docker --fail-swap-on=false --hostname-override=minikube
...
```

If you see that for `kubelet` the host network sources allowed is only `file` in flag `--host-network-sources=file`. So I changed that to allow anything by changing the file to `'*'`.

in file `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` change:

```
--host-network-sources=file  to  --host-network-sources='*'
```

and then to reflect those changes run:

```bash
systemctl daemon-reload
systemctl restart kubelet
```

And once that was done, I checked if the pod was started:

```console
$ kubectl get pods -n kube-system
NAME                               READY   STATUS    RESTARTS   AGE
...
kube-proxy-r28bc                   1/1     Running   0          69m
kube-scheduler-minikube            1/1     Running   0          69m
storage-provisioner                1/1     Running   0          69m
tiller-deploy-6d6cc8dcb5-5mrx9     1/1     Running   0          65m
```

So allow any host network source on minikube.

### The operating system name and version used

```console
$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

```console
$ uname -a
Linux foo.bar.io 4.20.3-042003-generic #201901171122 SMP Thu Jan 17 16:25:26 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

```console
$ minikube version
minikube version: v0.34.1
```

**Comment by tstromberg** [comment]
I wasn't able to replicate this in minikube v0.35.0 using your command-line, so I believe that we may have fixed the issue.

Do you mind confirming? If v0.35.0 does fail, please attach the output of `minikube logs`. Thanks!

**Comment by surajssd** [comment]
Yes this is fixed in `0.35.0`. I couldn't reproduce it either on the latest version.
